File: University Student Rating/Code/server/UniversityStudentRatingService/ServiceLogicLayer/Interfaces/IUserService.cs
using DataAccessLayer.DbContext;
using ServiceLogicLayer.DTO;
using ServiceLogicLayer.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLogicLayer.Interfaces
{
    public interface IUserService : IDisposable
    {
        IEnumerable<UserProfile> GetAll();
        UserProfile GetById(int id);
        IEnumerable<UserProfile> GetStudentUserProfiles(int universityId, int facultyId, int courseId);
        IEnumerable<UserProfile> GetStudentUserProfiles(int universityId, int facultyId, int courseId, int professorId, int discipineId);
        AuthUserDTO GetAuthUserProfile(string userName);
        bool SaveStudentProfessorDisciplines(SaveStudentProfessorDiscModel model);
        bool SaveProfessorDisciplines(SaveProfessorDisciplinesModel model);
        bool UpdateUserProfilePhotoPath(string userName, string photoPath);
    }
}

File: University Student Rating/Code/androidclient/UniversityStudentRating/app/src/main/java/com/example/arturrinkis/universitystudentrating/ServerServiceAPI/Interfaces/Services/IUserService.java
package com.example.arturrinkis.universitystudentrating.ServerServiceAPI.Interfaces.Services;

import com.example.arturrinkis.universitystudentrating.DTO.AuthUser;
import com.example.arturrinkis.universitystudentrating.DTO.SaveProfessorDisciplinesModel;
import com.example.arturrinkis.universitystudentrating.DTO.SaveStudentProfessorDistModel;
import com.example.arturrinkis.universitystudentrating.DTO.UserProfile;

import java.util.ArrayList;

public interface IUserService {
    UserProfile getUserProfileById(int id);
    ArrayList<UserProfile> getUserProfiles();
    ArrayList<UserProfile> getStudentUserProfiles(int universityId, int facultyId, int courseId);
    ArrayList<UserProfile> getStudentUserProfilesByProfDisc(int universityId, int facultyId, int courseId, int professorId, int disciplineId);
    AuthUser getAuthUser(boolean doLoadPhoto);
    boolean saveStudentProfessorDisciplines(SaveStudentProfessorDistModel model);
    boolean saveProfessorDisciplines(SaveProfessorDisciplinesModel model);
}
